### ISSUE TITLE
Bug report-change on mysql/psql flexible server results in error. Zone change is found when not wanted

### ISSUE THREAD

**Issue by BerendvW** [aztfmod/terraform-azurerm-caf#1405]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Version of the module you are using

5.5.5

### Rover Version

aztfmod/rover:1.2.9-2209.1506

### Affected Resource(s)/Data Source(s)

azurerm_mysql_flexible_server

### Terraform Configuration Files

```hcl
phpipam = {
    name                          = "phpipam"
    version                       = "8.0.21"
    sku_name                      = "GP_Standard_D2ds_v4"
    resource_group_key            = "phpipam_app"
    keyvault                      = { key = "kv_phpipam" }
    administrator_username        = "mysqladmin"
    public_network_access_enabled = false
    auto_grow_enabled             = false

    vnet = {
      key        = "phpipam"
      subnet_key = "mysql"
      lz_key     = "phpipam_network"
    }
    private_dns_zone = {
      key             = "dns_phpipam"
    }

    mysql_configurations = {
      mysql_configurations = {
        name  = "interactive_timeout"
        value = "600"
      }
    }

    mysql_databases = {
      mysql_database = {
        name      = "phpipam"
        charset   = "latin1"
        collation = "latin1_swedish_ci"
      }
    }

    backup_retention_days = 30

    maintenance_window = {
      day_of_week  = "6"
      start_hour   = "1"
      start_minute = "15"
    }
  }
```

### Expected Behaviour

No change has been made to the flexible server code. The state local and remote state should be the same and therefore no changes should be proposed.

### Actual Behaviour

The above server doesn't have a `high_availability` block. Azure does give it a zone when appplying initially.
The code looks for `high_availability` or `zone`. This is not set, so it defaults to null.
A change is planned/applied to change the zone from the state value to `null`.
```
  # module.solution.module.mysql_flexible_server["phpipam"].azurerm_mysql_flexible_server.mysql will be updated in-place
  ~ resource "azurerm_mysql_flexible_server" "mysql" {
        id                            = "/subscriptions/537af0ea-f855-4c3d-ada6-798b8244149d/resourceGroups/ssc-p-rg-phpipam_app/providers/Microsoft.DBforMySQL/flexibleServers/ssc-p-mysqlf-phpipam"
        name                          = "ssc-p-mysqlf-phpipam"
        tags                          = {
            "CostCenter"   = "0"
            "DR"           = "mission critical"
            "Env"          = "prd"
            "OpCo"         = "IT BV"
            "OpsTeam"      = "SSC"
            "Owner"        = "***"
            "ProductName"  = "phpipam"
            "ServiceClass" = "bronze"
            "module"       = "mysql_flexible_server"
        }
      - zone                          = "1" -> null
        # (15 unchanged attributes hidden)

        # (2 unchanged blocks hidden)
    }
```

### Steps to Reproduce

1. Apply the mysql_flexible_server code
2. After deployment, re-apply the same code

**Comment by BerendvW** [comment]
A workaround for this is:
set zone to your preferred zone (line 12) in `modules/databases/mysql_flexible_server/server.tf` e.g. `zone = try(var.settings.zone, 1)`
comment the high availability block in the same file:
```
  # dynamic "high_availability" {
  #   for_each = try(var.settings.high_availability, null) == null ? [] : [var.settings.high_availability]

  #   content {
  #     mode                      = "ZoneRedundant"
  #     standby_availability_zone = var.settings.zone == null ? null : var.settings.high_availability.standby_availability_zone
  #   }
  # }
```

**Comment by mitiko** [comment]
Note since zones are [shuffled](https://techcommunity.microsoft.com/t5/fasttrack-for-azure/is-your-availability-zone-quot-1-quot-equal-to-my-quot-1-quot/ba-p/3562349) per subscription (your zone 1 might physically be my zone 3), and since some locations don't have all availability zones compatible with MySQL/PostgreSQL flexible server deployments, hardcoding `zone = 1` won't work for everyone.  
To give an example, as time of writing, `eastus2` has only one availability zone compatible, and `westeurope` has two.

**Comment by mitiko** [comment]
I agree that the current workaround is to set a valid zone.  
If you skip setting the zone, Azure will choose one for you, and you can modify it before the re-apply.

However, I don't agree we should default to 1 in the module with `zone = try(var.settings.zone, 1)`.
If you set an invalid zone, you get the following much less informative error:

`
Error: waiting for creation of Flexible Server: (Name "xxxx-mysqlf-servername / Resource Group "xxxx-rg-rgname"): Code="InternalServerError" Message="An unexpected error occured while processing the request. Tracking ID: '00000000-1111-2222-3333-444444444444'
`

Here, even in the portal, for `eastus2` I only have zone 2 available for flexible servers' creation:
![limited-availability-zones](https://user-images.githubusercontent.com/26521741/226896056-fecb27b6-c710-4c6f-8960-c5955b98c64f.png)

**Comment by BerendvW** [comment]
@mitiko: I agree that the workaround should be whatever your preferred zone is (Which is why I put e.g. in the original post regarding this).
